pkgdb/src/buildenv/assets/activate.d/set-prompt.tcsh

```diff
@@ -0,0 +1,29 @@
+# Tweak the (already customized) prompt: add a flox indicator.
+if ( ! $?FLOX_PROMPT ) then
+    set FLOX_PROMPT = "flox"
+endif
+
+set colorReset = "%{\033[0m%}"
+set colorBold = "%{\033[1m%}"
+set colorPrompt1 = "%{\033[38;5;""$FLOX_PROMPT_COLOR_1""m%}"
+set colorPrompt2 = "%{\033[38;5;""$FLOX_PROMPT_COLOR_2""m%}"
+set _floxPrompt1 = "$colorPrompt1""$FLOX_PROMPT"
+set _floxPrompt2 = "$colorPrompt2""[$FLOX_PROMPT_ENVIRONMENTS]"
+
+if $?NO_COLOR then
+    set _flox = "flox [$FLOX_PROMPT_ENVIRONMENTS]"
+else
+    set _flox = "$colorBold$_floxPrompt1 $_floxPrompt2$colorReset"
+endif
+
+unset _esc colorReset colorBold colorPrompt1 colorPrompt2 _floxPrompt1 _floxPrompt2
+
+# Save the current 'tcsh_prompt' if not already saved.
+if ( $?prompt ) then
+    if ( ! $?FLOX_SAVE_TCSH_PROMPT ) then
+        setenv FLOX_SAVE_TCSH_PROMPT "$prompt"
+    endif
+    set prompt = "$_flox $FLOX_SAVE_TCSH_PROMPT"
+endif
+
+unset _flox
```

Comment:
suggestion blocking: when `"${FLOX_PROMPT_ENVIRONMENTS:-}" == ""`, the prompt shouldn't be changed at all

Reply:
Can we do that on a separate PR across all 4 shells, because that logic doesn't exist for bash or zsh either?

Reply:
I'm fine with a followup, but it does exist already, see https://github.com/flox/flox/blob/main/pkgdb/src/buildenv/assets/activate.d/set-prompt.bash#L25 and https://github.com/flox/flox/pull/1144

Reply:
Fixed in commit 0c470001a77b3fcb892f43062e5a532295a8a5f5